pkg/plugins/builtin/httputil/httputil.go

```diff
@@ -110,3 +113,42 @@ func defaultHTTPClientFactory(cfg HTTPClientConfig) HTTPClient {
 	}
 	return c
 }
+
+func GetTransport(opts ...func(*http.Transport)) http.RoundTripper {
+	tr := http.DefaultTransport.(*http.Transport).Clone()
+	for _, o := range opts {
+		o(tr)
+	}
+
+	_ = http2.ConfigureTransport(tr)
+	return tr
+}
+
+func WithTLSInsecureSkipVerify(v bool) func(*http.Transport) {
+	return func(t *http.Transport) {
+		if t.TLSClientConfig == nil {
+			t.TLSClientConfig = &tls.Config{}
+		}
+		t.TLSClientConfig.InsecureSkipVerify = v
+	}
+}
+
+func WithTLSClientAuth(cert tls.Certificate) func(*http.Transport) {
+	return func(t *http.Transport) {
+		if t.TLSClientConfig == nil {
+			t.TLSClientConfig = &tls.Config{}
+		}
+		t.TLSClientConfig.Certificates = []tls.Certificate{cert}
+	}
+}
+
+func WithTLSRootCA(caCert []byte) func(*http.Transport) {
+	return func(t *http.Transport) {
+		if t.TLSClientConfig == nil {
+			t.TLSClientConfig = &tls.Config{}
+		}
+		caCertPool := x509.NewCertPool()
```

Comment:
Same as for `WithTLSClientAuth`, if the cert poll already exist, maybe reuse it ?

Reply:
This method should not be re-used multiple time.
If someone want to alter the certPool with multiple certificates, I guess it should be done with one call to `WithTLSRootCA`, and providing multiples certificates as inputs.
Will clarify this in a comment.

Reply:
Yeah, I see where your're coming from on tat, but since the cert pool type allow to add certificates, does it really matter that we add to the existing pool instead of recreating one ?